internal/cliwrapper/cliwrapper_test.go

```diff
@@ -48,6 +49,54 @@ func TestPodman_ImageExists(t *testing.T) {
 	tests.RemoveImage(logger, tests.TestImage)
 }
 
+func TestPodman_ImageExists(t *testing.T) {
+	Podman_ImageExists(t, "")
+}
+
+func TestPodman_Remote_ImageExists(t *testing.T) {
+	var tmpPodmanSocketCmd *exec.Cmd
+
+	// Check if there is an existing connection of `podman-machine-default` since this is included when installing
+	// podman desktop for macOS.
+	connectionName := "podman-machine-default"
+	chkDefaultConnectionCmd := exec.Command(tests.GetPodmanPath(), "--connection", connectionName, "system", "info") //nolint:gosec
+	if err := chkDefaultConnectionCmd.Run(); err == nil {
+		Podman_ImageExists(t, connectionName)
+	} else if runtime.GOOS == "linux" {
```

Comment:
Apparently the linter thinks that the `else` clause is too complex for the embedded `err` construct.  Try changing this to
```go
	err := chkDefaultConnectionCmd.Run()
	if err == nil {
		Podman_ImageExists(t, connectionName)
	} else if runtime.GOOS == "linux" {
```

Reply:
You're simplifying the `if`, not the `else`. If the linter knows what it's talking about, the problem is the whole big Linux block below. You could instead stick with the `if` and simply return after doing the `Podman_ImageExists` then start a new `if runtime.GOOS == "linux"` block. Would that make it happier?

But, is it worth using the default Mac connection vs just always creating the temp service regardless of platform? (I think that "emergency cleanup" before the next run likely isn't an issue as the background service isn't being daemonized and ought to exit when the unit test process exits... but it wouldn't hurt to think about what happens if the socket or the connection definition already exists: for example, what if a system ends up running multiple copies at the same time? One solution would be to randomize the socket and connection names, for example by embedding a pid or a random UUID.)

Reply:
> Would that make it happier?

It's unhappy about the gross number of indentation levels (and, how it counts levels is slightly arcane).  In [my PR to Ed's branch](https://github.com/ewchong/arcaflow-engine-deployer-podman/pull/2), I ended up pulling most of the `else` body out into a helper function (because we needed to drop at least _two_ levels of indentation...).

> is it worth using the default Mac connection vs just always creating the temp service regardless of platform?

I think the challenge is figuring out how to create the temporary service on a platform other than `"linux"`.  Yeah, we can presumably do it on `"darwin"` _if_ Podman Desktop is installed, but it's much more straightforward to look for the "known"/"default" one and use it if it's there, and otherwise restrict us to where we know how to operate.  That, at least, lets things work in the CI.

Reply:
> what happens if the socket or the connection definition already exists

Yeah, I didn't really like having the socket created at the top level of `/var/tmp`...but I eventually punted on that.  Having it created in the current working directory might be good.  Or, putting it under `/run/user/${UID}` (which is where Podman and Docker seem to put theirs) might be nice, but that's very linux-y.  But, at the least, giving it a random filename would be a very good idea.